pandas/core/frame.py

```diff
@@ -8791,6 +8809,11 @@ def combiner(x: Series, y: Series):
         if dtypes:
             combined = combined.astype(dtypes)
 
+        combined = combined.reindex(columns=all_columns, fill_value=None)
+
+        if not sort_columns:
+            combined = combined[self.columns]
```

Comment:
Doesn't this drop columns if there are columns in `other` that aren't in `self`?